erpnext/accounts/report/financial_statements.py

```diff
@@ -417,13 +418,20 @@ def set_gl_entries_by_account(
 	gl_entries_by_account,
 	ignore_closing_entries=False,
 	ignore_opening_entries=False,
+	root_type=None,
 ):
 	"""Returns a dict like { "account": [gl entries], ... }"""
 	gl_entries = []
 
 	accounts_list = frappe.db.get_all(
 		"Account",
-		filters={"company": company, "is_group": 0, "lft": (">=", root_lft), "rgt": ("<=", root_rgt)},
+		filters={
+			"company": company,
+			"is_group": 0,
+			"lft": (">=", root_lft),
+			"rgt": ("<=", root_rgt),
+			"root_type": root_type,
```

Comment:
Root type should be passed as a parameter to filters only if has a value, passing None will give undesired results